library/std/src/panicking.rs

```diff
@@ -707,7 +705,7 @@ pub const fn begin_panic<M: Any + Send>(msg: M) -> ! {
 /// abort or unwind.
 fn rust_panic_with_hook(
     payload: &mut dyn BoxMeUp,
-    message: Option<&fmt::Arguments<'_>>,
+    message: Option<fmt::Arguments<'_>>,
```

Comment:
The message is now only used in case of must_abort, right? (And the payload is ignored in that case.) That seems worth mentioning in the doc comment.

Reply:
I've implemented `Display` for the `PanicPayload`s, so it now uses that instead of a separate message argument.